tools/wptserve/wptserve/request.py

```diff
@@ -262,22 +266,16 @@ def __init__(self, request_handler):
         self.url_base = "/"
 
         if self.request_path.startswith(scheme + "://"):
-            self.url = request_handler.path
+            self.url = self.request_path
         else:
-            self.url = "%s://%s:%s%s" % (scheme,
-                                      host,
-                                      port,
-                                      self.request_path)
+            self.url = "%s://%s:%s%s" % (
```

Comment:
This being a native string seems weird (also for other properties).

Reply:
I agree the inconsistency is a bit weird here, but I'm pretty sure changing this will break a bunch of things and should probably be another PR.

Reply:
OK, can you file an issue?

Reply:
```suggestion
            # TODO(#23362): Stop using native strings for URLs.
            self.url = "%s://%s:%s%s" % (
```